### ISSUE TITLE
Upload artifacts to Travis after CI builds

### ISSUE THREAD

**Issue by barbeau** [CUTR-at-USF/gtfs-realtime-validator#167]
**Summary:** 

I'd like to make current builds available to users to download without them having to build the project.

Looks like Travis has a way to upload artifacts to S3 following a build:
https://docs.travis-ci.com/user/uploading-artifacts/

**Comment by barbeau** [comment]
Using the default path config doesn't seem to upload the artifacts correctly to S3 when I have the bucket/key/secret entered into the Travis settings console.

When I add the environmental variable:

* `ARTIFACTS_PATHS` = `./target/gtfs-rt-validator-1.0-SNAPSHOT.jar`

...it uploads but to a build-specific path, for example:

* https://s3.amazonaws.com/gtfs-rt-validator/CUTR-at-USF/gtfs-realtime-validator/290/290.1/target/gtfs-rt-validator-1.0-SNAPSHOT.jar

We don't want that - we want the same URL to be used for the same JAR name.

**Comment by barbeau** [comment]
Here's where the build artifacts are generated on Travis (from https://api.travis-ci.org/jobs/226541505/log.txt?deansi=true):

~~~
/home/travis/.m2/repository/com/conveyal/gtfs-validation-lib/0.1.7-SNAPSHOT/resolver-status.properties
/home/travis/.m2/repository/com/conveyal/gtfs-validator/0.1.7-SNAPSHOT/resolver-status.properties
/home/travis/.m2/repository/com/conveyal/gtfs-validator-json/0.1.7-SNAPSHOT/resolver-status.properties
/home/travis/.m2/repository/edu/usf/cutr/gtfs-rt-validator/1.0-SNAPSHOT/gtfs-rt-validator-1.0-SNAPSHOT.jar
/home/travis/.m2/repository/edu/usf/cutr/gtfs-rt-validator/1.0-SNAPSHOT/maven-metadata-local.xml
/home/travis/.m2/repository/edu/usf/cutr/gtfs-rt-validator/1.0-SNAPSHOT/_remote.repositories
/home/travis/.m2/repository/edu/usf/cutr/gtfs-rt-validator/maven-metadata-local.xml
~~~

**Comment by barbeau** [comment]
More documentation specific to S3 at https://docs.travis-ci.com/user/deployment/s3/.

**Comment by barbeau** [comment]
This is a fully working example with the `deploy` config:
https://renzo.lucioni.xyz/s3-deployment-with-travis/

**Comment by barbeau** [comment]
Looks like `deploy` config is separate from `artifact` - [this build](https://travis-ci.org/CUTR-at-USF/gtfs-realtime-validator/builds/226568647) didn't upload artifact with this error:

~~~
No stash found.
missing access_key_id
failed to deploy
~~~

**Comment by barbeau** [comment]
With https://github.com/CUTR-at-USF/gtfs-realtime-validator/commit/d983bfa70e6b76ad7916caaa59e3173ecfc5d096, the upload to S3 is finally working.  According to S3, the publicly available URL is:
https://s3.amazonaws.com/gtfs-rt-validator/travis_builds/gtfs-rt-validator-1.0-SNAPSHOT.jar

...although when visiting there I currently see:

~~~
<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>6630A26A7BFEBCC6</RequestId><HostId>QH+dtAlvgs0qvRJzxMJM4HNJx6ZddafBI20g7jPa2iqDNdI4binuLCJeITr6aQksTI3snqCJ3aE=</HostId></Error>
~~~

I'll leave this issue open until I resolve the ACL issue.

**Comment by barbeau** [comment]
I think the ACL issue is fixed via AWS web console config, so I'm closing this.  I'll reopen if it's not.

**Comment by barbeau** [comment]
Nope, ACL issue still exists.

**Comment by barbeau** [comment]
Alright, ACL should be set to public read now via https://github.com/CUTR-at-USF/gtfs-realtime-validator/commit/c43d4c8f98be520d6133421393337f844c1113ac.

**Comment by barbeau** [comment]
Reopening, as I'm not sure the correct artifacts are being uploaded.  When trying to run the JAR and going to http://localhost:8080 I get a empty page with the message:

>HTTP ERROR: 404
>Problem accessing /. Reason:
>    Not Found
>Powered by Jetty://

Here's the directions for the README once we get the upload sorted:

~~~
## Quick Start

1. Install [Java Development Kit (JDK) 1.8 or higher](http://www.oracle.com/technetwork/java/javase/downloads/index-jsp-138363.html)
2. Download the latest alpha build:
    * [gtfs-rt-validator-1.0-SNAPSHOT.jar](https://s3.amazonaws.com/gtfs-rt-validator/travis_builds/gtfs-rt-validator-1.0-SNAPSHOT.jar)
3. From the command line run `java -jar gtfs-rt-validator-1.0-SNAPSHOT.jar`
4. When prompted, in your browser go to `http://localhost:8080`

## Building the project 
~~~

**Comment by barbeau** [comment]
Looks like artifact being downloaded from S3 generated by Travis is fine, it's the bug in #181 that's causing the 404 when running the JAR.  After we fix #181 we can add the above text to the README.
